app/experimenter/nimbus-ui/src/components/App/ExperimentRoot/index.test.tsx

```diff
@@ -0,0 +1,309 @@
+    const slug = "experiment-demo-slug";
+    const { mock, experiment } = mockExperimentQuery(slug, {
+      publishStatus: NimbusExperimentPublishStatusEnum.REVIEW,
+    });
+
+    const SubjectInner = (props: RouteComponentProps) => {
+      const { useRedirectCondition } = useContext(ExperimentContext)!;
+      useRedirectCondition(({ status }) => {
+        if (status.review) return "";
+      });
+      return <div data-testid="SubjectInner">Redirecting...</div>;
+    };
+
+    render(
+      <Subject {...{ slug, mocks: [mock] }}>
+        <SubjectInner path="edit" />
+      </Subject>,
+    );
+
+    await waitFor(() => {
+      expect(navigate).toHaveBeenCalledWith(`${BASE_PATH}/${experiment.slug}`, {
+        replace: true,
+      });
+    });
+  });
+
+  it("supplies a useExperimentPolling hook that enables experiment polling", async () => {
+    jest.useFakeTimers();
+
+    const slug = "experiment-demo-slug";
+    const { mock: initialMock, experiment: initialExperiment } =
+      mockExperimentQuery(slug);
+    const { mock: updatedMock, experiment: updatedExperiment } =
+      mockExperimentQuery(slug, {
+        publishStatus: NimbusExperimentPublishStatusEnum.WAITING,
+      });
+    const { mock: ignoredMock, experiment: ignoredExperiment } =
+      mockExperimentQuery(slug, {
+        publishStatus: NimbusExperimentPublishStatusEnum.APPROVED,
+      });
+
+    const SubjectInner = (props: RouteComponentProps) => {
+      const {
+        experiment: { publishStatus },
+      } = useContext(ExperimentContext)!;
+      const [enablePolling, setEnablePolling] = useState(false);
+      return (
+        <div data-testid="SubjectInner">
+          <ul>
+            <li data-testid="publishStatus">{publishStatus}</li>
+          </ul>
+          <button
+            data-testid="toggle-polling"
+            onClick={() => setEnablePolling((state) => !state)}
+          >
+            Toggle Polling
+          </button>
+          {enablePolling && <SubjectPollingEnabler />}
+        </div>
+      );
+    };
+
+    const SubjectPollingEnabler = () => {
+      const { useExperimentPolling } = useContext(ExperimentContext)!;
+      useExperimentPolling();
+      return <div data-testid="polling-enabled">Polling enabled</div>;
+    };
+
+    render(
+      <Subject {...{ slug, mocks: [initialMock, updatedMock, ignoredMock] }}>
+        <SubjectInner path="edit" />
+      </Subject>,
+    );
+
+    await screen.findByTestId("SubjectInner");
+    expect(screen.getByTestId("publishStatus")).toHaveTextContent(
+      initialExperiment!.publishStatus!,
+    );
+
+    // Toggling polling on should introduce SubjectPollingEnabler into the
+    // DOM and call useExperimentPolling as a side-effect, enabling polling.
+    fireEvent.click(screen.getByTestId("toggle-polling"));
+    await waitFor(() => {
+      expect(screen.queryByTestId("polling-enabled")).toBeInTheDocument();
+    });
```

Comment:
Is testing this explicit UI toggling strictly necessary?  I don't think we toggle it anywhere in the actual UI, is just enabling it for this dummy component and doing the fast forward and checking that it's called sufficient?

Reply:
This is kind of a simulation for how polling works in the app. 

Navigating to the experiment summary causes PageSummary to appear in the document, which enables polling via `useEffect` in `useExperimentPolling`. Navigating away to another page like PageEditOverview causes the cleanup handler from `useEffect` to run and disable polling. 

The `SubjectPollingEnabler` component just exercises `useExperimentPolling` in isolation. Waiting for the `polling-enabled` element to appear / disappear just ensures all the side effects have happened before manipulating the timer